### ISSUE TITLE
Check that atom map does not result in ring formation/breaking

### ISSUE THREAD

**Issue by pgrinaway** [choderalab/perses#438]
We should brainstorm a way to check that the atom map does not result in ring formation or breaking.

**Comment by jchodera** [comment]
On reflection, this should be easy with `networkx`:
* Create a graph from the initial molecular topology
* Enumerate all cycles in the initial molecular graph
* For each cycle, iterate over edges and check that the final molecular topology has a corresponding bond. If not, we've broken a ring

**Comment by jchodera** [comment]
That reminds me that retrieving `networkx` representations are something we might want to add to our new `Topology` and `Molecule` classes:
```python
nxtop = topology.to_networkx()
nxmol = molecule.to_networkx()
```

**Comment by jchodera** [comment]
Slight modification of the algorithm for efficiency:
* Create a graph from the initial molecular topology
* [Enumerate all cycles](https://networkx.github.io/documentation/stable/reference/algorithms/cycles.html?highlight=cycle#module-networkx.algorithms.cycles) in the initial molecular graph
* For each cycle, iterate over edges and add the the mapped pair of atoms in the final topology (if they exist) to a `set`
* Ensure all atom pairs in the `set` are connected by a bond in the final topology

**Comment by jchodera** [comment]
Two more observations:
* We probably have to check both directions: ring bonds in A are also bonds in B (if the atoms are mapped), *and* ring bonds in B are also bonds in A (if the atoms are mapped)
* We can use the much simpler `oechem` [`IsInRing`](https://docs.eyesopen.com/toolkits/python/oechemtk/OEChemClasses/OEBondBase.html#OEChem::OEBondBase::IsInRing) bond method to enumerate bonds in rings.

**Comment by jchodera** [comment]
Fixed in #444.
